schema/query_test.go

```diff
@@ -158,7 +171,7 @@ func TestQueryInvalidType(t *testing.T) {
 	_, err = ParseQuery("{\"foo\": {\"$gt\": 1}}", s)
 	assert.EqualError(t, err, "foo: cannot apply $gt operation on a non numerical field")
 	_, err = ParseQuery("{\"bar\": {\"$gt\": \"1\"}}", s)
-	assert.EqualError(t, err, "bar: value for $gt must be a number")
+	assert.EqualError(t, err, "bar: value for $gt must be a number or time")
```

Comment:
It must be of the same type as the field. A query comparing time against number should not be possible.